libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -121,3 +143,108 @@ queryDRepStakeDistr nes creds
 queryCommitteeState :: NewEpochState era -> CommitteeState era
 queryCommitteeState nes =
   vsCommitteeState $ certVState $ lsCertState $ esLState $ nesEs nes
+{-# DEPRECATED queryCommitteeState "In favor of `queryCommitteeMembersState`" #-}
+
+-- | Query committee members
+queryCommitteeMembersState ::
+  forall era.
+  EraGov era =>
+  -- | filter by cold credentials (don't filter when empty)
+  Set (Credential 'ColdCommitteeRole (EraCrypto era)) ->
+  -- | filter by hot credentials (don't filter when empty)
+  Set (Credential 'HotCommitteeRole (EraCrypto era)) ->
+  -- | filter by status (don't filter when empty)
+  -- (useful, for discovering, for example, only active members)
+  Set MemberStatus ->
+  NewEpochState era ->
+  CommitteeMembersState (EraCrypto era)
+queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes =
```

Comment:
Current implementation of this query is a bit complicated and it is hard to follow. I suggest that it will be easier to start with a set of relevant cold keys and convert it to a Map with some filtering, something along these lines:
```haskell

      relevantHotCreds :: Set (Credential 'ColdCommitteeRole (EraCrypto era))
      relevantHotCreds
        | null hotCredsFilter = comStateMembers
        | otherwise = Map.filter (maybe False (`Set.member` hotCredsFilter)) comStateMembers

      relevantColdCreds :: Set (Credential 'ColdCommitteeRole (EraCrypto era))
      relevantColdCreds
        -- we only look for Unrecognized members when it is contained in the filter, or when there is no filtering
        | Set.null statusFilter || Set.member Unrecognized statusFilter =
            Map.keysSet comMembers `Set.union` Map.keysSet relevantHotCredsMap
        | otherwise = Map.keysSet comMembers

      filteredColdCreds
        | null coldCredsFilter = relevantColdCreds
        | otherwise = relevantColdCreds `Map.restrictKeys` coldCredsFilter

      committeMemeberState = 
        Map.mapMaybe id $ Map.fromSet mkMaybeMemberState filteredColdCreds

      mkMaybeMemberState :: Credential 'ColdCommitteeRole (EraCrypto era) ->
                            Maybe (CommitteeMemberState (EraCrypto era))
      mkMaybeMemberState coldCred = do
        let status =
              case Map.lookup coldCred comMembers of
                Nothing -> Unrecognized
                Just member | nes ^. nesELL > expirationEpoch member -> Expired
                            | otherwise -> Active
        guard (null statusFilter || status `Set.member` statusFilter)
        ...
```
This way you can focus on a single cold credential at a time in `mkMaybeMemberState`, which only needs to filter by status and you can use `do` syntax for Maybe to keep it readable.

Reply:
Nice suggestion, thank you. 
I shuffled it around, because in this form it's  a little wrong (the `union` after the `filter` overwrites the filter), but I kept the same idea.